### PR TITLE
Support .json?dcr for interactives

### PR DESCRIPTION
**Pull request by nicl**
*Will compress the commits before merging.*

## What does this change?

Adds support for `.json?dcr` on interactives. This also paves the way for `?dcr` on interactives, though that requires a little DCR work too.

Note, as part of this work a *lot* of refactoring was done:

* moving constructors into companion objects for DCR models to improve readability of the main `for*` methods by reducing clutter, and also because it is a standard Scala convention to do so. Note, this also includes the DCR constructor itself which now lives in the `DotcomRenderingDataModel` companion rather than in the `DotcomRenderingUtils` helper object
* separate out liveblog from article special handling (branch early rather than matching within the main method)
* a small amount of renaming

### Tested

More local testing is required to make sure I haven't broken anything in the model.

- [x] Locally
- [ ] On CODE (optional)

### CODE REVIEW

**Review by shtukas — APPROVED** [review]

**Comment by nicl** [comment]
Update, have tested on 50 articles locally against `main` and can't see meaningful diffs so bit more confident in this now.

File: applications/app/controllers/InteractiveController.scala [hunk 1]
```diff
@@ -66,21 +63,28 @@ class InteractiveController(
...
+  )(implicit request: RequestHeader): Future[Either[(InteractivePage, Blocks), Result]] = {
     val edition = Edition(request)
     log.info(s"Fetching interactive: $path for edition $edition")
     val response: Future[ItemResponse] = contentApiClient.getResponse(
       contentApiClient
         .item(path, edition)
         .showFields("all")
-        .showAtoms("all"),
+        .showAtoms("all")
+        .showBlocks("all"),
```
Comment (jfsoul) [inline applications/app/controllers/InteractiveController.scala:76]: I suspect you could use CAPILookup.lookup for this now as it's basically the same lookup we do for any content now.

File: common/app/model/dotcomrendering/DotcomRenderingDataModel.scala [hunk 2]
```diff
@@ -173,4 +163,248 @@ object DotcomRenderingDataModel {
...
+        oldest = paginationInfo.oldest.map(_.suffix),
+        older = paginationInfo.older.map(_.suffix),
+      )
+    })
+
+    val bodyBlocks = DotcomRenderingUtils.blocksForLiveblogPage(page, blocks)
+    val keyEvents =
+      blocks.requestedBodyBlocks
+        .getOrElse(Map.empty[String, Seq[APIBlock]])
+        .getOrElse("body:key-events", Seq.empty[APIBlock])
```
Comment (jfsoul) [inline common/app/model/dotcomrendering/DotcomRenderingDataModel.scala:232]: Not  sure why but this stumped me for a bit, `Option.getOrElse` followed by `Map.getOrElse`

**Comment by jfsoul** [comment]
Nice one, pretty cool that we have some semblance of sensible data model sharing/splitting for articles, liveblogs and interactives in DCR now!

**Pull request merged**